stripped/colabs/Birch_stripped.py

```diff
@@ -0,0 +1,732 @@
+X_scaled.head()
+
+
+# In[ ]:
+
+
+plot_2d_tsne_for_three_different_perplexities([10,30,50], X_scaled, czy_step_to_pustynia=True)
+
+
+# In[ ]:
+
+
+from sklearn.linear_model import LogisticRegression
+
+
+# In[ ]:
+
+
+tsne = TSNE(n_components = 2, perplexity = 50, learning_rate=10000, random_state=2024, n_jobs=-1)
+tsne_results = tsne.fit_transform(X_scaled.loc[:,'Rainf':'SoilM_100_200cm'])
+
+
+# In[ ]:
+
+
+df = pd.DataFrame()
+df['dim 1'] = tsne_results[:,0]
+df['dim 2'] = tsne_results[:,1]
+sns.scatterplot(x="dim 1", y="dim 2", palette = ["darkgreen", "orange", "blue"], data=df, legend="full")
+
+
+# In[ ]:
+
+
+df = pd.DataFrame()
+df['dim 1'] = tsne_results[:,0]
+df['dim 2'] = tsne_results[:,1]
+df['Cluster'] = X_scaled["pustynia_i_step"]
+
+
+# In[ ]:
+
+
+clf = LogisticRegression(random_state=0).fit(df[["dim 1", "dim 2"]], nasa_anotated_500_202307['pustynia_i_step'])
+
+
+# In[ ]:
+
+
+df['Cluster pred'] = clf.predict(df[["dim 1", "dim 2"]])
+
+
+# In[ ]:
+
+
+sns.scatterplot(x="dim 1", y="dim 2", hue="Cluster", palette = ["darkgreen", "orange", "blue"], data=df, legend="full").set(ylim=(min(df['dim 2']-1), max(df['dim 2']+1)), xlim=(min(df['dim 1']-1), max(df['dim 1']+1)), title="Regresja logistyczna")
+b = clf.intercept_[0]
+w1, w2 = clf.coef_.T
+
+c = -b/w2
+m = -w1/w2
+
+xmin, xmax = min(df['dim 1']-1), max(df['dim 1']+1)
+ymin, ymax = min(df['dim 2']-1), max(df['dim 2']+1)
+xd = np.array([xmin, xmax])
+yd = m*xd + c
+plt.plot(xd, yd, 'k', lw=1, ls='--')
+plt.fill_between(xd, yd, ymin, color='tab:green', alpha=0.2)
+plt.fill_between(xd, yd, ymax, color='tab:orange', alpha=0.2)
+
+
+# In[ ]:
+
+
+from sklearn.metrics import classification_report
+print(classification_report(clf.predict(df), nasa_anotated_500_202307['pustynia_i_step']))
+
+
+# Do grupowania będziemy używać następujących wskaźników:
+# 
+# *   **Rainf** - wskaźnik opadów deszczu,
+# *   **Evap** - wskaźnik całkowitej ewapotranspiracji,
+# *   **AvgSurfT** - wskaźnik średniej temperatury powierzchni ziemi,
+# *   **Albedo** - wskaźnik albedo,
+# *   **SoilT_40_100cm** - wskaźnik temperatury gleby w warstwie o głębokości od 40 do 100 cm,
+# *   **GVEG** - wskaźnik roślinności,
+# *   **PotEvap** - wskaźnik potencjalnej ewapotranspiracji,
+# *   **RootMoist** - wilgotność gleby w strefie korzeniowej (parowanie, które miałoby miejsce, gdyby dostępne było wystarczające źródło wody),
+# *   **SoilM_100_200cm** - wilgotność gleby w warstwie o głębokości od 100 do 200 cm.
+
+# In[ ]:
+
+
+# Elementy list_of_scaled_df są w kolejności chronologicznej tzn. list_of_scaled_df[0] to dane z 10.2022, a list_of_scaled_df[11] to dane z 09.2023.
+list_of_scaled_df = []
+for dataframe in list_of_df:
+  X = dataframe.loc[:,'Rainf':'SoilM_100_200cm']
+  X_scaled = MinMaxScaling(StandardScaling(X, X.columns), X.columns)
+  X_scaled = pd.concat([dataframe[["lon", "lat"]], X_scaled], axis=1)
+  list_of_scaled_df.append(X_scaled)
```

Comment:
Chyba się dublujemy tutaj niepotrzebnie? Czy to ta sama list_of_df co wcześniej? (nie widzę innej)